packages/radix-ui-themes/src/helpers/require-react-element.ts

```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+
+/** A function that throws an error when a value isn't a valid React Element, otherwise returns the value */
+export const requireReactElement = <T extends React.ReactNode>(children: T): T => {
+  const isReactElement = React.isValidElement(children);
+
+  if (!isReactElement) {
+    throw Error(
+      `Expected a single React Element child, but got: ${React.Children.toArray(children)
+        .map((child) =>
+          typeof child === 'object' && 'type' in child && typeof child.type === 'string'
+            ? child.type
+            : typeof child
+        )
+        .join(', ')}`
+    );
+  }
```

Comment:
Would be nice to have this in the Slot primitive itself

Reply:
I'm confused by this a little, it seems to replicate what `React.Children.Only` would do already? Don't you get this message (or very similar) anyway if you pass more than one child?

I've tried and if I pass nothing in a dialog trigger it fails silently, and if I pass 2 buttons, I still get the error from `React.Children.Only`, so when am I supposed to see this?

Reply:
This isn't really about ensuring that a single child is passed, more about ensuring an element is passed as a child, and not string / number / null / etc

Pedro and iirc someone else complained that it's unexpected how triggers like `<DialogTrigger>Open</DialogTrigger>` don't work and also don't throw any errors, so it looks just like a silent bug if you expected it to render a button like the primitives do